### DISCUSSION TITLE
Empty subscriptions

### DISCUSSION THREAD

**Discussion by ajouve** [hyochan/react-native-iap#1983] [General]
I tried to create a React Native project a year ago but having [an issue I could not solve](https://github.com/dooboolab/react-native-iap/issues/1231) with the in-app purchase I made a pause, and now I am coming back, but I have a similar issue 

The issue was in android, and I am testing with real devices.

I created a new Google play console account and generated a subscription, my identity and the bank account has been verified.

<img width="1678" alt="Screenshot 2022-09-21 at 11 55 19" src="https://user-images.githubusercontent.com/2337492/191475532-bf524141-a9d0-4e1f-be0b-ddf9be8b01c3.png">

This is my actual subscription configuration, let's say that my product ID is `sub1` and the only plan is `plan1`

`plan1` is available in the country I am testing

Then I have four google account in the device I am doing tests, I added all of them in `License testing` and the response is `RESPOND_NORMALLY`

Now the code

```
import React, { useEffect } from 'react';
import { SafeAreaView, Image, Text, View, TouchableOpacity } from 'react-native';
import { PurchaseError, requestSubscription, useIAP } from 'react-native-iap';

function Subscription({ navigation }) {

    const { connected, subscriptions, getSubscriptions } = useIAP();

    useEffect(function () {
        if(connected) {
            handleGetSubscriptions();
        }
    }, [connected])

    async function handleButtonCLick() {
        console.log('subscriptions', subscriptions);
        console.log('connected', connected);
    }

    async function handleGetSubscriptions() {
        const subscriptionSkus = Platform.select({
            ios: [],
            android: ['sub1', 'plan1'],
            default: [],
        })
        try {
            console.log('skus', subscriptionSkus);
            await getSubscriptions({ skus: subscriptionSkus });
        } catch (error) {
            console.error(error)
        }
    }

    return (
        <SafeAreaView  >
            <View  >
                <Text>{JSON.stringify(subscriptions)}</Text>
                <TouchableOpacity onPress={handleButtonCLick} >
                    <Text >Test</Text>
                </TouchableOpacity>
            </View>
        </SafeAreaView>
    );
};
```

Having a look at my logs, `handleGetSubscriptions` is called, and I have no error
When clicking on the button, I have `subscriptions []` and `connected true`
And the `Text` with the `subscriptions` is always `[]`

Not sure if for the sku I have to use the `sub1` or `plan1` but using both I have an empty array, I also tried with only `sub1` and only `plan1`, same issue

I am running `npm run android` to test

From time to time I have this error `[Error: An unknown or unexpected error has occurred. Please try again later.]` I have no more details, I do not know if this error is related to `react-native-iap` sometime I can test during hours without seeing this error and sometimes I have this error some seconds after each launch

**Comment by andresesfm** [comment]
@ajouve you cannot test this on dev mode. You need to build a release build and at least upload it to the play store to complete the setup

**Reply by ajouve** [reply]
Thanks for the help, it took days for the product to be available

**Reply by RickYangzz** [reply]
![image](https://github.com/dooboolab-community/react-native-iap/assets/38999212/c28be633-fedd-4980-93fc-8058465b0c97)
I'd like to know if this internal status(not reviewed) is ok for me to test react-native-iap?

**Comment by baby230211** [comment]
@andresesfm  
Same issue here, i am using the same code there 
AND it only happens on Android Subscription, instead of iOS products, subscription, Android products
i can get products using getProducts, but it's not working on getSubscriptions.
It return empty array i think 

i had confirmed that the skus and the country  is correct

any help?

**Comment by andresesfm** [comment]
@baby230211 you have to use the same build number and have your app published to the play store for this to work. If you created the subs on a version previous to the use you are using locally, the play store will not return those

**Reply by baby230211** [reply]
this is the steps i follows, does any setup looks wrong?
1. create subscription as following images
2. upload release .aab (includes ```<uses-permission android:name="com.android.vending.BILLING"/>```) to google play console 
3. enter https://play.google.com/apps/testing/`${android_packageName}` to install app on android device

i think i'm using the same build number and have published to the play store
btw, i found that i haven't link to the google cloud project, is this may be problems of not showing subscriptions?

**Reply by baby230211** [reply]
i found out that my subscription on google play console is non-renewable, 
i create one renewable subscription, then it shows on the app!!

**Comment by andresesfm** [comment]
https://stackoverflow.com/questions/11068686/this-version-of-the-application-is-not-configured-for-billing-through-google-pla
